src/Ghcid.hs

```diff
@@ -382,8 +385,9 @@ runGhcid session waiter termSize termOutput opts@Options{..} = do
             whenLoud $ outStrLn $ "%RELOADING: " ++ unwords reason
             restartTimes2 <- mapM getModTime restart
             let restartChanged = [s | (False, s) <- zip (zipWith (==) restartTimes restartTimes2) restart]
+                restartFilesChanged = or [(ra ++ [pathSeparator]) `isPrefixOf` rb | ra <- absRestart, rb <- reason]
```

Comment:
See `addTrailingPathSeparator`.

Reply:
This line is pretty subtle - can you include a link to https://github.com/ndmitchell/ghcid/issues/273 so the reasoning is clear.